src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,210 @@
+        List<Car> cars = new ArrayList<>();
+        String query = "SELECT c.id AS car_id, c.model, m.id AS manufacturer_id, m.name, m.country "
+                + "FROM cars c JOIN cars_drivers cd ON c.id = cd.car_id "
+                + "JOIN manufacturers m ON m.id = c.manufacturer_id "
+                + "WHERE c.is_deleted = FALSE AND cd.driver_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCarWithManufacturer(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get cars by driver id: " + driverId, e);
+        }
+        cars.forEach(car -> car.setDrivers(getCarDrivers(car.getId())));
+        return cars;
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Long carId = resultSet.getObject("cars.id", Long.class);
+        String carModel = resultSet.getObject("cars.model", String.class);
+        String manufacturerName = resultSet.getObject("m.name", String.class);
+        String manufacturerCountry = resultSet.getObject("m.country", String.class);
+        Long manufacturerId = resultSet.getObject("cars.manufacturer_id", Long.class);
+        Manufacturer manufacturer = new Manufacturer(manufacturerName, manufacturerCountry);
+        manufacturer.setId(manufacturerId);
+        Car car = new Car(carModel, manufacturer);
+        car.setId(carId);
+        return car;
+    }
+
+    private List<Driver> getCarDrivers(Long id) {
+        String query = "SELECT d.id, d.name, d.license_number FROM drivers d "
+                + "INNER JOIN cars_drivers cd ON cd.driver_id = d.id "
+                + "WHERE car_id = ? AND d.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            List<Driver> drivers = new ArrayList<>();
+            while (resultSet.next()) {
+                drivers.add(getDriver(resultSet));
+            }
+            return drivers;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get drivers by car id: " + id, e);
+        }
+    }
+
+    private Driver getDriver(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getObject("id", Long.class);
+        String name = resultSet.getString("name");
+        String country = resultSet.getString("license_number");
+        return new Driver(id, name, country);
+    }
+
+    private Car getCarWithManufacturer(ResultSet resultSet) throws SQLException {
```

Comment:
we have two methods (getCarWithManufacturer and getCar) with the same logic, let's remove one of them